integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/loading/SearchQueryEntityLoadingMultipleTypesIT.java

```diff
@@ -227,6 +231,52 @@ public void mixedHierarchies() {
 		);
 	}
 
+	@Test
+	public void mixedHierarchies_entityLoadingTimeout() {
```

Comment:
Same here: you need to simulate slow loading, at least.

Also, I'm sure this test currently passes (from time to time) only because your implementation rounds the timeout value to the nearest second.